src/map/pc.c

```diff
@@ -9014,6 +9030,11 @@ static int pc_itemheal(struct map_session_data *sd, int itemid, int hp, int sp)
 		if (sd->sc.data[SC_BITESCAR]) {
 			hp = 0;
 		}
+		
+		if (sd->sc.data[SC_NO_RECOVER_STATE]) {
+			hp -= hp;
+			sp -= sp;
```

Comment:
Any reason to do `hp -= hp` instead of just `hp = 0` (and the same for sp)? I think that'd be more clear to the reader